python/test/unit/language/test_core.py

```diff
@@ -3683,6 +3700,8 @@ def kernel(
     if in_dtype_str == 'int8':
         out = numpy_random((B, M, N), dtype_str='int32', rs=rs)
     else:
+        x *= 0.1
+        y *= 0.1
```

Comment:
Precision is not enough for some fp16xfp16->fp16 tests. Without this and FMA path I get errors like this:
```
FAILED python/test/unit/language/test_core.py::test_dot3d[1-1-64-64-64-32-32-float16-float16] - AssertionError: 
Not equal to tolerance rtol=0.01, atol=0.01

Mismatched elements: 9 / 4096 (0.22%)
Max absolute difference: 0.0781
Max relative difference: 0.833
 x: array([[[ -2.621 ,  10.03  , -13.05  , ...,   9.05  , -24.95  ,
          16.52  ],
        [-13.04  ,  15.22  , -11.414 , ...,   6.58  , -14.41  ,...
 y: array([[[ -2.627 ,  10.04  , -13.03  , ...,   9.05  , -24.97  ,
          16.48  ],
        [-13.02  ,  15.21  , -11.4   , ...,   6.562 , -14.414 ,...

```

By making gemm arguments smaller we effectively make absolute tolerance larger.
I copied this trick from test_dot.

I am sure this is not a compiler issue, because I've implemented gemm using bunch of numpy data casts and strict accumulation order to emulate code generated by Triton to generate reference output and got no mismatch errors:

``` python
  if out_dtype_str == 'float16':
    for b in range(B):
        for m in range(M):
            for n in range(N):
                out_ref[b, m, n] = 0.0
                for k in range(K):
                    out_ref[b, m, n] = np.float32(out_ref[b, m, n]) + np.float32(x[b, m, k]) * np.float32(y[b, k, n])
```

Results are still not bitwise equal, but it seems the difference somewhere in the last 1-2 bits now.

Reply:
I think Nvidia backend could be affected by this problem as well(not yet, because for Nvidia is disabled in this PR), since we loose precision with FMA a lot. Unfortunately I can not check it at the moment, but will update it in a follow up PRs.

Reply:
I think we can still gate it behind hip for now to be conservative. We can broaden it later if we enable the nvidia path. I pushed 4db0218 for this.